Lib/test/libregrtest/single.py

```diff
@@ -184,6 +187,11 @@ def _runtest_env_changed_exc(result: TestResult, runtests: RunTests,
         result.state = State.PASSED
 
 
+class PrintWarningsType(Protocol):
+    orig_stderr: io.TextIOWrapper
+    def __call__(self, msg: str) -> None: ...
```

Comment:
If possible, I would prefer avoiding to have to add glue on top of code just to please mypy. If print_warning() API is too weird for mypy, maybe rewrite it with such class directly in test.support, no?

Reply:
> Mypy doesn't know that the test.support module exists

Can't we make mypy aware of test.support, instead of having to put ducktape to workaround these limitations?

Reply:
> Can't we make mypy aware of test.support

This is somewhat tricky due to the fact that `regrtest` is in the `Lib/test/` directory. Mypy hates being run on stuff inside the `Lib/` directory, as it thinks that everything inside the `Lib/` directory is "shadowing the stdlib". The only way we're managing to get mypy happy with being run on `libregrtest` is by `cd`-ing into `Lib/test/`, and then running mypy with very specific options so that it doesn't look anywhere else in the `Lib/` directory, it _only_ looks in `Lib/test/libregrtest`.

There _would_ be some ways of making mypy aware of `test.support`, but I'm not sure I like any of them:

1. We could write a custom script that copies `libregrtest` and `test.support` into a temporary directory, then runs mypy on them together. **Why I don't like this:** it's easier for users if they can run mypy just by invoking `mypy` on the command line. Having to run mypy via a wrapper script is a bad user experience, and it's more code for _us_ to maintain as well.
2. We could add stubs for `test.support` in typeshed, so that mypy sees `test.support` as "just another part of the stdlib". **Why I don't like this:** lots of Python users can't access `test.support` on their installed versions of Python, but if we add stubs for `test.support` in typeshed, mypy will never complain if users import `test.support` in their own code. Also, it will be a lot of work for us typeshed maintainers to maintain stubs for `test.support`, since the `test.support` module tends to change its API a lot more frequently than other things in the `Lib/` directory.
3. We could add stubs for `test.support` inside the `libregrtest` directory itself. **Why I don't like this:** this just feels like it would be really ugly, and really hard to maintain. The stubs would go out of date really quickly, if we weren't careful.

Reply:
> We could write a custom script that copies libregrtest and test.support into a temporary directory

That sounds like a good idea to me.

I would prefer that mypy users need workaround, than libregrtest have to write complicated workarounds in the working code just to please mypy.